stripe/api_requestor.py

```diff
@@ -151,38 +151,88 @@ def request(self, method, url, params=None, headers=None):
         resp = self.interpret_response(rbody, rcode, rheaders)
         return resp, my_api_key
 
-    def handle_api_error(self, rbody, rcode, resp, rheaders):
+    def handle_error_response(self, rbody, rcode, resp, rheaders):
         try:
-            err = resp['error']
+            error_data = resp['error']
         except (KeyError, TypeError):
             raise error.APIError(
                 "Invalid response object from API: %r (HTTP response code "
                 "was %d)" % (rbody, rcode),
                 rbody, rcode, resp)
 
+        err = None
+        if isinstance(error_data, basestring):
```

Comment:
Maybe just add a comment here on why we're doing this. I think that the minor structural difference between an OAuth and API error is pretty subtle unless you already know about it.